### ISSUE TITLE
Fix S4261: False positive on async Main

### ISSUE THREAD

**Issue by jincod** [SonarSource/sonar-csharp#1644]
### Description

A false positive S4261 is thrown on the line:

### Repro steps

```csharp
static async Task Main(string[] args)
```

### Expected behavior

Don't suggest to rename method Main

### Actual behavior

![image](https://user-images.githubusercontent.com/1294432/43888176-58489f3c-9bda-11e8-8c22-3559ee34a792.png)

### Related information

* SonarC# Version 7.3 (build 5690)
* SonarScanner for MSBuild 4.1.1.1164 - .NET Fwk 4.6 (Jenkins plugin)

**Comment by Evangelink** [comment]
Hi @jincod,

Thanks for the feedback! We will fix this FP in the current sprint.

Cheers,
Amaury
